avm/client/src/avmHelpers.ts

```diff
@@ -0,0 +1,129 @@
+
+const decoder = new TextDecoder();
+const encoder = new TextEncoder();
+
+export function prepareArgs(
+    air: string,
+    prevData: Uint8Array,
+    data: Uint8Array,
+    params: {
+        initPeerId: string;
+        currentPeerId: string;
+    },
+    callResults: CallResultsArray,
+): string {
+    const callResultsToPass: any = {};
+    for (let [k, v] of callResults) {
+        callResultsToPass[k] = {
+            ret_code: v.retCode,
+            result: v.result,
+        };
+    }
+
+    const paramsToPass = {
+        init_peer_id: params.initPeerId,
+        current_peer_id: params.currentPeerId,
+    };
+
+    const encoded = encoder.encode(JSON.stringify(callResultsToPass));
+
+    const avmArg = JSON.stringify([
+        // force new line
+        air,
+        Array.from(prevData),
+        Array.from(data),
+        paramsToPass,
+        Array.from(encoded),
+    ]);
+
+    return avmArg;
+}
+
+export function convertInterpreterResult(rawResult: string): InterpreterResult {
+    let result: any;
+    try {
+        result = JSON.parse(rawResult);
+    } catch (ex) {
+        throw 'call_module result parsing error: ' + ex + ', original text: ' + rawResult;
+    }
+
+    if (result.error !== '') {
+        throw 'call_module returned error: ' + result.error;
+    }
+
+    result = result.result;
+
+    const callRequestsStr = decoder.decode(new Uint8Array(result.call_requests));
+    let parsedCallRequests;
+    try {
+        if (callRequestsStr.length === 0) {
+            parsedCallRequests = {};
+        } else {
+            parsedCallRequests = JSON.parse(callRequestsStr);
+        }
+    } catch (e) {
+        throw "Couldn't parse call requests: " + e + '. Original string is: ' + callRequestsStr;
+    }
+
+    let resultCallRequests: Array<[key: number, callRequest: CallRequest]> = [];
+    for (const k in parsedCallRequests) {
+        const v = parsedCallRequests[k];
+
+        let arguments_;
+        let tetraplets;
+        try {
+            arguments_ = JSON.parse(v.arguments);
+        } catch (e) {
+            throw "Couldn't parse arguments: " + e + '. Original string is: ' + arguments_;
+        }
+
+        try {
+            tetraplets = JSON.parse(v.tetraplets);
+        } catch (e) {
+            throw "Couldn't parse tetraplets: " + e + '. Original string is: ' + tetraplets;
+        }
+
+        resultCallRequests.push([
+            k as any,
+            {
+                serviceId: v.service_id,
+                functionName: v.function_name,
+                arguments: arguments_,
+                tetraplets: tetraplets,
+            },
+        ]);
+    }
+    return {
+        retCode: result.ret_code,
+        errorMessage: result.error_message,
+        data: result.data,
+        nextPeerPks: result.next_peer_pks,
+        callRequests: resultCallRequests,
+    };
+}
+
+type FaaSCall = ((args: string) => Promise<string>) | ((args: string) => string);
+
+export async function runAvm(
```

Comment:
Maybe make only this function exported from this file? It's highly likely that only it would be used from external code.

Reply:
And let's add a comment about this function.

Reply:
I don't see any harm in exporting additional pure functions. They won't break anything, but might become useful if there is any change we'll need to implement a different method of passing args to the interpreter

Reply:
There is no any harm in exporting these functions, but it's just basic encapsulation principles. F.e. these public could confuse me if I read this code for the first time.